### PR TITLE
update to redux-toolkit

### PR DESCRIPTION
**Pull request by gadgetfather**
- I have removed authContext and replaced it with redux-toolkit
- Users can signup, login, and logout 
### Test credentials :
username: 123
password: 123

### CODE REVIEW

File: src/components/Toast/Toast.jsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { toast } from "react-toastify";
+export const toastSuccess = (message) => {
+  toast.success(message, {
+    position: "top-center",
+    autoClose: 2000,
+    hideProgressBar: false,
+    closeOnClick: true,
+    pauseOnHover: true,
+    draggable: true,
+    progress: undefined,
```
Comment (Jay-Gadhiya) [inline src/components/Toast/Toast.jsx:10]: you can extract props into a new file and give them everywhere,

**Review by Jay-Gadhiya — APPROVED** [review]
- code written well
- UI looks good

File: src/pages/LoginPage/LoginPage.jsx [hunk 2]
```diff
@@ -1,28 +1,32 @@
...
-      login(formData);
+      dispatch(userLogin(formData));
     }
   };
+
+  useEffect(() => {
+    if (token && location.pathname === "/") {
+      navigate("/home");
+    }
+  }, [token, navigate, location]);
```
Comment (starver20) [inline src/pages/LoginPage/LoginPage.jsx:29]: Nice detail, the user won't be able to stay on the login page if he's logged in.

**Review by starver20 — APPROVED** [review]
Well written auth slice.
Liked the attention given to small details.
Good to merge

**Pull request merged**